### ISSUE TITLE
Ask a question 

### ISSUE THREAD

**Issue by skyblue** [mweststrate/immer#228]
Please specify:

(Not filling out _all fields_ of this template will probably result in your issue being closed without further notice. If you are not willing fill it in, why would someone else spent his time fixing it?)

* [ ] Feature request: ______
  * [ ] I am willing to implement this myself
* [x] Issue: _______
  * Version: ____
  * _SIMPLE_ Reproduction: ____ (tip: fork this sandbox: https://codesandbox.io/s/vmxmn8omwl)
  * Expected behavior: ___
  * Observed behavior: ___
  * [ ] Occurs when using Proxies (use `setUseProxies(true)`)
  * [ ] Occurs in the ES5 implementation (use `setUseProxies(false)`)
  

```javascript
 const state = [
    {
      todo: 'Learn typescript',
      done: true
    },
    {
      todo: 'Try immer',
      done: false
    }
  ]

  const nextState = produce(state, draft => {
    draft.push({todo: 'Tweet about it'})
    draft[1].done = true
  })

  nextState[1].todo = 'Modify directly'
  console.log(nextState)
  ```

nextState[1] not change,  It's correct?

**Comment by aleclarson** [comment]
You're wondering why setting `nextState[1].todo` outside the `produce` callback does nothing?

Everything outside the `produce` callback is immutable.

**Comment by aleclarson** [comment]
> Everything in the nextState object is immutable outside the produce callback.

This is actually **unintended.** Objects created inside a producer should _never_ be auto-frozen.

Fixed by #231

**Comment by aleclarson** [comment]
Oh wait, I misread your code. I thought you were mutating `nextState[2]` outside the producer, not `nextState[1]`!

You did `draft[1].done = true` in the producer, which causes `nextState[1] !== state[1]` and `Object.isFrozen(nextState[1])` **as is intended.** Object copies created by Immer are auto-frozen by default in development. The only way to disable this is with `setAutoFreeze(false)`.

Thus, you cannot do `nextState[1].todo =` outside the producer, but you _can_ do `nextState[2].todo =` once #231 is merged.
